modules/apps/27-interchain-accounts/host/types/params_legacy.go

```diff
@@ -0,0 +1,58 @@
+/*
+NOTE: Usage of x/params to manage parameters is deprecated in favor of x/gov
+controlled execution of MsgUpdateParams messages. These types remains solely
+for migration purposes and will be removed in a future release.
+*/
+
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+var (
+	// KeyHostEnabled is the store key for HostEnabled Params
+	KeyHostEnabled = []byte("HostEnabled")
+	// KeyAllowMessages is the store key for the AllowMessages Params
+	KeyAllowMessages = []byte("AllowMessages")
+)
+
+// ParamKeyTable type declaration for parameters
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+// ParamSetPairs implements params.ParamSet
+func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyHostEnabled, &p.HostEnabled, validateEnabledType),
+		paramtypes.NewParamSetPair(KeyAllowMessages, &p.AllowMessages, validateAllowlistLegacy),
+	}
+}
+
+func validateEnabledType(i interface{}) error {
+	_, ok := i.(bool)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	return nil
+}
+
+func validateAllowlistLegacy(i interface{}) error {
```

Comment:
Are `validateEnabledType` and `validateAllowlistLegacy` still necessary? My understanding is that only the interface functions need to be kept around for migrations?

Reply:
You are right, they are not needed.

Reply:
they are actually needed. See lines 31 and 32. `x/params` module expects a validator function for each registered param.

Reply:
However, I don't know if we **actually** need them since params "should" already be registered if we're doing a migration. But maybe we need to do this each time the chain is restarted. Keeping them certainly makes testing much easier. I think we should keep them since this file is ought to be removed anyway.

Reply:
oh yea, lets leave it then :+1: